controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +552,116 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS records for deleted services/pods
+			dnsRecord := &infranetworkv1.DNSData{}
+			dnsName := "dns-" + serviceName + fmt.Sprintf("-%d", i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			err = helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsRecord)
+			if err != nil && !k8s_errors.IsNotFound(err) {
+				Log.Info(fmt.Sprintf("Error while getting DNS object: %v", err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while cleaning the DNS record %s: %v", dnsName, err))
+			}
+
 		}
 	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Since DNS will always return ovsdbserver-xb.openstack.svc... this fields
+	// are an string instead of an array of strings. The add and removal of entries
+	// will be done using the DNSData CR, but this fields will remain the same on scale-up/down
+	dbAddress := ""
+	internalDbAddress := ""
+	raftAddress := ""
+
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
+		// Get Hostname
+		svc, err := service.GetServiceWithName(
+			ctx,
+			helper,
+			ovnPod.Name,
+			ovnPod.Namespace,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+		hostnames = append(hostnames, hostname)
+
+		// Get IP
+		net_stat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+		if err != nil {
+			Log.Info(fmt.Sprintf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err))
+			return ctrl.Result{}, err
+		}
+		for _, v := range net_stat {
+			if v.Interface == instance.Spec.NetworkAttachment {
+				// Return only IPv4 at the moment
+				if strings.Count(v.IPs[0], ":") == 0 {
+					dnsIP = v.IPs[0]
+					break
+				}
+			}
+		}
+
+		// Create DNSRecord
+		var DNSRecords []infranetworkv1.DNSHost
+		// ovsdbserver-xb-x entry
+		DNSRecord := infranetworkv1.DNSHost{}
+		DNSRecord.IP = dnsIP
+		DNSRecord.Hostnames = hostnames
+		// ovsdbserver-xb entry
+		headless_dns_hostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+		if instance.Spec.DBType == v1beta1.NBDBType {
+			headless_dns_hostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
+		}
+		dbAddress = fmt.Sprintf("tcp:%s:%d", headless_dns_hostname, svc.Spec.Ports[0].Port)
+		internalDbAddress = fmt.Sprintf("tcp:%s.cluster.local:%d", headless_dns_hostname, svc.Spec.Ports[0].Port)
```

Comment:
For internalDbAddress and raftaddresses may be keep using non headless services? i.e for 3 replicas it will be 3 dns addresses and let them configuered at original position. There is some commented logic for dns part, can use/fix that itself.
And here handle only the dbaddress part and that too only when networkattachment != ""

Reply:
This is currently what I'm working on, internalDbAddress needs to be configured right after service is created, as currently, as you say, if networkAttachment == "" it won't be set (which is wrong).

Reply:
Was having doubts regarding which would be best if using headless service (ovsdbserver-sb.namespace.svc.cluster.local) or using service one (ovsdbserver-sb-[n].namespace.svc.cluster.local).
After some test regarding ping, coreDNS doesn't use RR in order to resolve name, doing:
`ping ovsdbserver-sb.openstack.svc.cluster.local -c1` it was always resolved to `ovsdbserver-sb-2.ovsdbserver-sb.openstack.svc.cluster.local`. I'll change InternalDbAddress and raftAddress to use the service names instead.